diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/identifiers/LocalVariablesRule.kt

```diff
@@ -0,0 +1,160 @@
+    }
+
+    private fun handleConsecutiveDeclarations(statement: PsiElement, properties: List<KtProperty>) {
+        // need to check that properties are declared consecutively with only maybe empty lines
+        properties
+                .sortedBy { it.node.lineNumber()!! }
+                .zip(properties.size - 1 downTo 0)
+                .forEach { (property, offset) ->
+                    checkLineNumbers(property, statement.node.lineNumber()!!, offset)
+                }
+    }
+
+    private fun checkLineNumbers(property: KtProperty, firstUsageStatementLine: Int, offset: Int = 0, firstUsageLine: Int? = null) {
+        val numLinesToSkip = property
+                .siblings(forward = true, withItself = false)
+                .takeWhile { it is PsiWhiteSpace || it.node.isPartOfComment() }
+                .let { siblings -> siblings.last().node.lastLineNumber()!! - siblings.first().node.lineNumber()!! - 1 }
+
+        if (firstUsageStatementLine - numLinesToSkip != property.node.lastLineNumber()!! + 1 + offset) {
+            LOCAL_VARIABLE_EARLY_DECLARATION.warn(configRules, emitWarn, isFixMode,
+                    warnMessage(property.name!!, property.node.lineNumber()!!, firstUsageLine
+                            ?: firstUsageStatementLine), property.startOffset)
+        }
+    }
+
+    /**
+     * Returns the [KtBlockExpression] with which a property should be compared.
+     * @return either the line on which the property is used if it is first used in the same scope, or the block in the same scope as declaration
+     */
+    private fun getFirstUsageStatementOrBlock(usages: List<KtNameReferenceExpression>, declarationScope: KtBlockExpression): PsiElement {
+        val firstUsage = usages.minBy { it.node.lineNumber()!! }!!
+        val firstUsageScope = firstUsage.getParentOfType<KtBlockExpression>(true)
+
+        return if (firstUsageScope == declarationScope) {
+            // property is first used in the same scope where it is declared, we check line of statement where it is first used
+            firstUsage
+                    .parents
+                    .find { it.parent == declarationScope }!!
+        } else {
+            // first usage is in deeper block compared to declaration, need to check how close is declaration to the first line of the block
+            usages.minBy { it.node.lineNumber()!! }!!
+                    .parentsWithSelf
+                    .find { it.parent == declarationScope }!!
+        }
+    }
+
+    private fun KtCallExpression?.isWhitelistedMethod() =
+            this?.run {
+                (referenceExpression() as KtNameReferenceExpression).getReferencedName() in functionInitializers &&
+                        valueArguments.isEmpty()
+            } ?: false
+
+    private fun warnMessage(name: String, declared: Int, used: Int) = "$name is declared on line $declared and used for the first time on line $used"
```

Comment:
and "is" used, please also separate line number and name by some separators like "<>" or "[]"